python/cuvs/cuvs/preprocessing/quantize/scalar/scalar.pyx

```diff
@@ -0,0 +1,232 @@
+#
+# Copyright (c) 2025, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# cython: language_level=3
+
+import numpy as np
+
+from cuvs.common cimport cydlpack
+
+from pylibraft.common import auto_convert_output, device_ndarray
+from pylibraft.common.cai_wrapper import wrap_array
+
+from cuvs.common.exceptions import check_cuvs
+from cuvs.common.resources import auto_sync_resources
+from cuvs.neighbors.common import _check_input_array
+
+
+cdef class QuantizerParams:
+    """
+    Parameters for scalar quantization
+
+    Parameters
+    ----------
+    quantile: float
+        specifies how many outliers at top & bottom will be ignored
+        needs to be within range of (0, 1]
+    """
+
+    cdef cuvsScalarQuantizerParams * params
+
+    def __cinit__(self):
+        check_cuvs(cuvsScalarQuantizerParamsCreate(&self.params))
+
+    def __dealloc__(self):
+        check_cuvs(cuvsScalarQuantizerParamsDestroy(self.params))
+
+    def __init__(self, *, quantile=None):
+        if quantile is not None:
+            self.params.quantile = quantile
+
+    @property
+    def quantile(self):
+        return self.params.quantile
+
+
+cdef class Quantizer:
+    """
+    Defines and stores scalar for quantisation upon training
+
+    The quantization is performed by a linear mapping of an interval in the
+    float data type to the full range of the quantized int type.
+    """
+    cdef cuvsScalarQuantizer * quantizer
+
+    def __cinit__(self):
+        check_cuvs(cuvsScalarQuantizerCreate(&self.quantizer))
+
+    def __dealloc__(self):
+        check_cuvs(cuvsScalarQuantizerDestroy(self.quantizer))
+
+    @property
+    def min(self):
+        return self.quantizer.min_
+
+    @property
+    def max(self):
+        return self.quantizer.max_
+
+    def __repr__(self):
+        return f"scalar.Quantizer(min={self.min}, max={self.max})"
+
+
+@auto_sync_resources
+def train(QuantizerParams params, dataset, resources=None):
+    """
+    Initializes a scalar quantizer to be used later for quantizing the dataset.
+
+    Parameters
+    ----------
+    params : QuantizerParams object
+    dataset : row major dataset
```

Comment:
Here too we should be specifying whether it can exist on decive, host, or both (ideally we should be able to support both).